packages/lesswrong/lib/types/schemaTypes.ts

```diff
@@ -28,24 +28,43 @@ type FieldName<N extends CollectionNameString> = (keyof ObjectsByCollectionName[
 
 type SqlFieldFunction<N extends CollectionNameString> = (fieldName: FieldName<N>) => string;
 
+type SqlJoinType = "inner" | "full" | "left" | "right";
+
 type SqlJoinBase<N extends CollectionNameString> = {
   table: N,
-  type?: "inner" | "full" | "left" | "right",
+  type?: SqlJoinType,
   on: Partial<Record<FieldName<N>, string>>,
 }
 
 type SqlResolverJoin<N extends CollectionNameString> = SqlJoinBase<N> & {
+  isNonCollectionJoin?: false,
```

Comment:
I think this could use a bit of documentation in the code - like I will probably forget that I need to set this flag